signer/signer.py

```diff
@@ -1,143 +1,163 @@
+                        current_block_no
+                    ],
+                )
+                if not config["private_key"].startswith("0x"):
+                    config["private_key"] = "0x" + config["private_key"]
+                signature = bytes(
+                    w3.eth.account.signHash(defunct_hash_message(message),
+                                            config["private_key"]).signature)
                 signature = signature.hex()
                 if not signature.startswith("0x"):
                     signature = "0x" + signature
-                return {"snet-free-call-user-id": username, "snet-payment-channel-signature-bin": signature,
-                        "snet-current-block-number": current_block_no, "snet-payment-type": "free-call"}
-            else:
-                raise Exception(
-                    "Free calls expired for username %s.", username)
-        except Exception as e:
-            print(repr(e))
-            raise e
-
-    def _get_user_address(self, username):
-        try:
-            wallet_address_data = self.repo.execute("SELECT address FROM wallet WHERE username = %s AND status = 1 "
-                                                    "LIMIT 1", [username])
-            if len(wallet_address_data) == 1:
-                return wallet_address_data[0]['address']
-            raise Exception(
-                "Unable to find wallet address for username %s", username)
-        except Exception as e:
-            print(repr(e))
-            raise e
-
-    def _get_service_metadata(self, org_id, service_id):
-        """
-            Method to get group details for given org_id and service_id.
-        """
-        try:
-            result = self.repo.execute(
-                "SELECT O.*, G.* FROM org_group O, service_group G WHERE O.org_id = G.org_id AND G.group_id = "
-                "O.group_id AND G.service_id = %s AND G.org_id = %s AND G.service_row_id IN (SELECT row_id FROM service "
-                "WHERE is_curated = 1 ) LIMIT 1", [org_id, service_id])
-            if len(result) == 1:
-                metadata = result[0]
-                payment = json.loads(metadata["payment"])
-                pricing = json.loads(metadata["pricing"])
-                metadata.update(payment)
-                metadata["pricing"] = pricing
-                return metadata
+                return {
+                    "snet-free-call-user-id": username,
+                    "snet-payment-channel-signature-bin": signature,
+                    "snet-current-block-number": current_block_no,
+                    "snet-payment-type": "free-call",
+                }
             else:
-                raise Exception(
-                    "Unable to find service for service_id %s", service_id)
+                raise Exception("Free calls expired for username %s.",
+                                username)
         except Exception as e:
-            print(repr(e))
+            logger.info(repr(e))
```

Comment:
exception should belogger as logger.exception /logger.error , please make necessary change at every place